model.py

```diff
@@ -51,33 +53,21 @@ def extract_args(func):
         return inspect.getargspec(func)[0]
 
 
-def shape_hints(**hints):
-    assert hints, "hints can not be empty"
-    assert all(isinstance(h, (list, tuple)) for h in hints.values()), \
-        "shape hint must be a list or tuple"
+class Input(fluid.dygraph.Layer):
```

Comment:
what is the point of making it a class instead of a plain function?

Reply:
In order to add data Variable into Program when only calling forward in static graph.  If a function to call fluid.data(), needs to clone var.